lua/entities/gmod_wire_expression2/core/custom/cfc_vector2_NumberOperands.lua

```diff
@@ -0,0 +1,537 @@
+end
+
+e2function vector2 cos(vector2 vectah)
+    return {
+        cos(vectah[1]*deg2rad),
+        cos(vectah[2]*deg2rad)
+    }
+end
+e2function vector2 vector2:cos()
+    return {
+        cos(this[1]*deg2rad),
+        cos(this[2]*deg2rad)
+    }
+end
+
+e2function vector2 sec(vector2 vectah)
+    return {
+        1/cos(vectah[1]*deg2rad),
+        1/cos(vectah[2]*deg2rad)
+    }
+end
+e2function vector2 vector2:sec()
+    return {
+        1/cos(this[1]*deg2rad),
+        1/cos(this[2]*deg2rad)
+    }
+end
+
+e2function vector2 sin(vector2 vectah)
+    return {
+        sin(vectah[1]*deg2rad),
+        sin(vectah[2]*deg2rad)
+    }
+end
+e2function vector2 vector2:sin()
+    return {
+        sin(this[1]*deg2rad),
+        sin(this[2]*deg2rad)
+    }
+end
+
+e2function vector2 csc(vector2 vectah)
+    return {
+        1/sin(vectah[1]*deg2rad),
+        1/sin(vectah[2]*deg2rad)
+    }
+end
+e2function vector2 vector2:csc()
+    return {
+        1/sin(this[1]*deg2rad),
+        1/sin(this[2]*deg2rad)
+    }
+end
+
+e2function vector2 tan(vector2 vectah)
+    return {
+        tan(vectah[1]*deg2rad),
+        tan(vectah[2]*deg2rad)
+    }
+end
+e2function vector2 vector2:tan()
+    return {
+        tan(this[1]*deg2rad),
+        tan(this[2]*deg2rad)
+    }
+end
+
+e2function vector2 cot(vector2 vectah)
+    return {
+        1/tan(vectah[1]*deg2rad),
+        1/tan(vectah[2]*deg2rad)
+    }
+end
+e2function vector2 vector2:cot()
+    return {
+        1/tan(this[1]*deg2rad),
+        1/tan(this[2]*deg2rad)
+    }
+end
+
+e2function vector2 rcos(vector2 vectah)
+    return {
+        cos(vectah[1]),
+        cos(vectah[2])
+    }
+end
+e2function vector2 vector2:rcos()
+    return {
+        cos(this[1]),
+        cos(this[2])
+    }
+end
+
+e2function vector2 rsec(vector2 vectah)
+    return {
+        1/cos(vectah[1]),
+        1/cos(vectah[2])
+    }
+end
+e2function vector2 vector2:rsec()
+    return {
+        1/cos(this[1]),
+        1/cos(this[2])
+    }
+end
+
+e2function vector2 rsin(vector2 vectah)
+    return {
+        sin(vectah[1]),
+        sin(vectah[2])
+    }
+end
+e2function vector2 vector2:rsin()
+    return {
+        sin(this[1]),
+        sin(this[2])
+    }
+end
+
+e2function vector2 rcsc(vector2 vectah)
+    return {
+        1/sin(vectah[1]),
+        1/sin(vectah[2])
+    }
+end
+e2function vector2 vector2:rcsc()
+    return {
+        1/sin(this[1]),
+        1/sin(this[2])
+    }
+end
+
+e2function vector2 rtan(vector2 vectah)
+    return {
+        tan(vectah[1]),
+        tan(vectah[2])
+    }
+end
+e2function vector2 vector2:rtan()
+    return {
+        tan(this[1]),
+        tan(this[2])
+    }
+end
+
+e2function vector2 rcot(vector2 vectah)
+    return {
+        1/tan(vectah[1]),
+        1/tan(vectah[2])
+    }
+end
+e2function vector2 vector2:rcot()
+    return {
+        1/tan(this[1]),
+        1/tan(this[2])
+    }
+end
+
+e2function vector2 acosrvec2(vector2 vectah)
```

Comment:
```suggestion
e2function vector2 acosr(vector2 vectah)
```

Reply:
I have to do this....

Reply:
I have to have the vec2.

Reply:
I have to add it to not get it to conflict with another one of the same name that takes a vector 2 and returns a number.

Reply:
I suggest you read the following function